aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -122,6 +122,9 @@ std::vector<Tensor> where(const Tensor& condition) {
 }
 
 Tensor _s_where_cpu(const Tensor& condition, const Tensor& self, const Tensor& other) {
+  if (self.dtype() != other.dtype()) {
```

Comment:
nit: it's better to just write `TORCH_CHECK(self.dtype() == other.dtype(), ...`.

Reply:
Changed